### PR TITLE
cli: take URL from Workspace also in absolute references

### PR DESCRIPTION
**Pull request by sttts**
Take the right URL from the workspace object for e.g. `kubectl ws root:example`.

Note: this does not contain a solution for the root workspace yet. Probably we need a similar solution to the `~` workspace where we fake a `Workspace` object.

### CODE REVIEW

File: pkg/cliplugins/workspace/plugin/kubeconfig.go [hunk 1]
```diff
@@ -217,15 +217,24 @@ func (kc *KubeConfig) UseWorkspace(ctx context.Context, name string) (err error)
 			return fmt.Errorf("current URL %q does not point to cluster workspace", config.Host)
 		}
 
-		if strings.Contains(name, ":") && logicalcluster.New(name).HasPrefix(tenancyv1alpha1.RootCluster) {
+		if strings.Contains(name, ":") && cluster.HasPrefix(currentClusterName) {
```
Comment (sttts) [inline pkg/cliplugins/workspace/plugin/kubeconfig.go:220]: hmm, bad completion?

File: pkg/cliplugins/workspace/plugin/kubeconfig.go [hunk 2]
```diff
@@ -217,15 +217,24 @@ func (kc *KubeConfig) UseWorkspace(ctx context.Context, name string) (err error)
 			return fmt.Errorf("current URL %q does not point to cluster workspace", config.Host)
 		}
 
-		if strings.Contains(name, ":") && logicalcluster.New(name).HasPrefix(tenancyv1alpha1.RootCluster) {
+		if strings.Contains(name, ":") && cluster.HasPrefix(tenancyv1alpha1.RootCluster) {
+			// absolute logical cluster under root:
 			parentClusterName, workspaceName := logicalcluster.New(name).Split()
-			if _, err := kc.personalClient.Cluster(parentClusterName).TenancyV1beta1().Workspaces().Get(ctx, workspaceName, metav1.GetOptions{}); err != nil {
+			ws, err := kc.personalClient.Cluster(parentClusterName).TenancyV1beta1().Workspaces().Get(ctx, workspaceName, metav1.GetOptions{})
```
Comment (p0lyn0mial) [inline pkg/cliplugins/workspace/plugin/kubeconfig.go:223]: ok, this is the case we hti today.

File: pkg/cliplugins/workspace/plugin/kubeconfig.go [hunk 3]
```diff
@@ -217,15 +217,24 @@ func (kc *KubeConfig) UseWorkspace(ctx context.Context, name string) (err error)
...
+			// intentionally do not check for readiness here
+
+			newServerHost = ws.Status.URL
+			workspaceType = &ws.Spec.Type
+		} else if strings.Contains(name, ":") {
+			// e.g. system:something
+			u.Path = path.Join(u.Path, logicalcluster.New(name).Path())
+			newServerHost = u.String()
+		} else if name == tenancyv1alpha1.RootCluster.String() {
+			// root workspace
```
Comment (p0lyn0mial) [inline pkg/cliplugins/workspace/plugin/kubeconfig.go:237]: it looks exactly the same as the previous case

Comment (p0lyn0mial) [reply pkg/cliplugins/workspace/plugin/kubeconfig.go:237]: clarified: in the future, the root case will require additional logic

**Pull request merged**